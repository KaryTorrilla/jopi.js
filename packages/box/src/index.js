import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
