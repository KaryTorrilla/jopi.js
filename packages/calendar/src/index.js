import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
