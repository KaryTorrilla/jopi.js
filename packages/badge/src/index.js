import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
