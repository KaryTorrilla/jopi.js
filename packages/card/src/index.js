import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
