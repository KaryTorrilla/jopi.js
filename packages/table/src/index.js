import React from 'react'
import { Box } from '@oneloop/box'

export const Table = (props) => (
  <Box
    as="table"
    {...props}
    __css={{
      fontFamily: 'primary',
      width: '100%',
      borderSpacing: '0 4px',
    }}
  />
)
/*
const TableHeaderItem = (props) => (
  <Box
    as="th"
    {...props}
    __css={{
      py: '16px',
      color: 'white',
      mb: '5px',
      backgroundColor: 'secondary',
      // ':firstchild': {
      //   borderRadius: 'input',
      // },
    }}
  />
)*/

const TableBody = (props) => <Box as="tbody" {...props} />
/*
const TableHeader = ({ children, ...props }) => (
  <Box as="thead">
    <Box as="tr" {...props}>
      {children}
    </Box>
  </Box>
)
const TableRow = (props) => <Box as="tr" {...props} />

const TableRowItem = (props) => (
  <Box
    as="td"
    {...props}
    __css={{
      py: '16px',
      color: 'darkGrey',
      mb: '5px',
      backgroundColor: 'inputBackground',
      // ':firstchild': {
      //   borderRadius: 'input',
      // },
    }}
  />
)*/

const TableHeader = ({ children, ...props }) => (
  <Box as="thead">
    <Box
      as="tr"
      {...props}
      __css={{
        fontFamily: 'Nunito Sans',
        padding: '12px 16px 12px 16px',
        gap: '8px',
        backgroundColor: 'neutralGray2',
        borderRadius: '8px',
        display: 'table-row',
        flexDirection: 'row',
        alignItems: 'center',
        mb: '8px',
        height: '60px',
      }}
      >
      {children}
    </Box>
  </Box>
)

const TableHeaderItem = ({ children, ...props }) => (
  <Box
    as="th"
    {...props}
    __css={{
      color: 'white',
      py: '12px',
      backgroundColor: 'neutralGray2',
      '&:first-child': {
        paddingLeft: '12px',
        borderBottomLeftRadius: '8px',
        borderTopLeftRadius: '8px',
      },
      '&:last-child': {
        paddingRight: '12px',
        borderBottomRightRadius: '8px',
        borderTopRightRadius: '8px',
      },
    }}
  >
    <Box
      __css={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
      }}
    >
      {children}
    </Box>
  </Box>
)

const TableRow = ({ children, disabled, variant = 'primary', selected, pressed, error, infoAlert, ...props }) => (
  <Box
    as="tr"
    tx='rows'
    variant={variant}
    {...props}
    __css={{
      fontFamily: 'Nunito Sans',
      padding: '6px 12px 6px 12px',
      gap: '12px',
      backgroundColor: '#FFFFFF',
      borderRadius: '8px',
      height: '54px',
      mb: '2px',
      color: disabled ? 'neutralGray4' : 'neutralGray2',
    }}
    >
    {children}
  </Box>
)

const TableRowItem = ({ children, error, ...props }) => (
  <Box
    as="td"
    tx='rows'
    variant= {error && 'error'}
    {...props}
    __css={{
      gap: '1px',
      '&:first-child': {
        paddingLeft: '12px',
        borderBottomLeftRadius: '8px',
        borderTopLeftRadius: '8px',
      },
      '&:last-child': {
        paddingRight: '12px',
        borderBottomRightRadius: '8px',
        borderTopRightRadius: '8px',
      },
    }}
  >
    <Box
      __css={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
      }}
    >
      {children}
    </Box>
  </Box>
)

Table.Header = TableHeader
Table.HeaderItem = TableHeaderItem
Table.Rows = TableBody
Table.Row = TableRow
Table.RowItem = TableRowItem
/*
Table.HeaderDefault = TableHeaderDefault
Table.HeaderItemDefault = TableHeaderItemDefault
Table.RowDefault = TableRowDefault
Table.RowItemDefault = TableRowItemDefault
*/
