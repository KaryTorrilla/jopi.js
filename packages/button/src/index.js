import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
