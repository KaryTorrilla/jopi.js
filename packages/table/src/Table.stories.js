import React from 'react'
import { Badge } from '@oneloop/badge'
import { Checkbox } from '@oneloop/checkbox'
import { Image } from '@oneloop/image'
import { Table } from '.'
import { Text } from '@oneloop/text'

export default {
  component: Table,
  title: 'Table',
}

export const normal = () => (
  <Table>
    <Table.Header>
      <Table.HeaderItem>Agency</Table.HeaderItem>
      <Table.HeaderItem>Country</Table.HeaderItem>
      <Table.HeaderItem>Paid</Table.HeaderItem>
    </Table.Header>
    <Table.Rows>
      <Table.Row>
        <Table.RowItem>InmoMex</Table.RowItem>
        <Table.RowItem>Mexico</Table.RowItem>
        <Table.RowItem>Yes</Table.RowItem>
      </Table.Row>
      <Table.Row>
        <Table.RowItem>RGM</Table.RowItem>
        <Table.RowItem>Argentina</Table.RowItem>
        <Table.RowItem>Yes</Table.RowItem>
      </Table.Row>
      <Table.Row>
        <Table.RowItem>Urbania</Table.RowItem>
        <Table.RowItem>Peru</Table.RowItem>
        <Table.RowItem>Yes</Table.RowItem>
      </Table.Row>
    </Table.Rows>
  </Table>
)

export const secondary = () => (
  <div style={{ background: '#E4E8EA', padding: '20px', borderRadius: '10px' }}>
    <Table>
      <Table.HeaderDefault>
        <Table.HeaderItemDefault></Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Foto</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Cód. Ref</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Dirección</Text>
          <Text variant="white.1">Ubicación</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Valor</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Dorm</Text>
          <Text variant="white.1">Baños</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">S. Cubierta</Text>
          <Text variant="white.1">Total</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">Fondo</Text>
          <Text variant="white.1">Frente</Text>
        </Table.HeaderItemDefault>
        <Table.HeaderItemDefault>
          <Text variant="white.0">S. Descrub</Text>
          <Text variant="white.1">S. Descrub</Text>
        </Table.HeaderItemDefault>
      </Table.HeaderDefault>
      <Table.Rows>
        <Table.RowDefault>
          <Table.RowItemDefault>
            <Checkbox />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
          <Image
            src="https://images.pexels.com/photos/1133957/pexels-photo-1133957.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
            variant="rows"
          />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">IHO4362866</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12Ellipsis" width='150px'>Av. Libertador 5687 5to A - Lote 1232</Text>
            <Text variant="bodyBold.fontSize11">Departamento en Palermo</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">Venta</Text>
            <Text variant="bodyBold.fontSize11">USD 200.000</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">2 Dorm</Text>
            <Text variant="body.fontSize11">2 Baños</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">87m° Cub</Text>
            <Text variant="body.fontSize11">0m° Tot</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">3 Amb</Text>
            <Text variant="body.fontSize11">0.00</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">120m° Semi</Text>
            <Text variant="body.fontSize11">0m° Total</Text>
          </Table.RowItemDefault>
        </Table.RowDefault>
        <Table.RowDefault>
          <Table.RowItemDefault>
            <Checkbox />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
          <Image
            src="https://images.pexels.com/photos/1133957/pexels-photo-1133957.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
            variant="rows"
          />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">IHO4362866</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">Av. Libertador 5687 5to A</Text>
            <Text variant="bodyBold.fontSize11">Departamento en Palermo</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">Venta</Text>
            <Text variant="bodyBold.fontSize11">USD 200.000</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">2 Dorm</Text>
            <Text variant="body.fontSize11">2 Baños</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">87m° Cub</Text>
            <Text variant="body.fontSize11">0m° Tot</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">3 Amb</Text>
            <Text variant="body.fontSize11">0.00</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">120m° Semi</Text>
            <Text variant="body.fontSize11">0m° Total</Text>
          </Table.RowItemDefault>
        </Table.RowDefault>
        <Table.RowDefault>
          <Table.RowItemDefault>
            <Checkbox />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
          <Image
            src="https://images.pexels.com/photos/1133957/pexels-photo-1133957.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
            variant="rows"
          />
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">IHO4362866</Text>
            <Text variant="body.fontSize11">AP4468985</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">Av. Libertador 5687 5to A</Text>
            <Text variant="bodyBold.fontSize11">Departamento en Palermo</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">Venta</Text>
            <Text variant="bodyBold.fontSize11">USD 200.000</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">2 Dorm</Text>
            <Text variant="body.fontSize11">2 Baños</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">87m° Cub</Text>
            <Text variant="body.fontSize11">0m° Tot</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Text variant="body.fontSize12">3 Amb</Text>
            <Text variant="body.fontSize11">0.00</Text>
          </Table.RowItemDefault>
          <Table.RowItemDefault>
            <Badge variant="badgeTips">COHIM</Badge>
            <Text variant="body.fontSize11">0m° Total</Text>
          </Table.RowItemDefault>
        </Table.RowDefault>
      </Table.Rows>
    </Table>
  </div>
)