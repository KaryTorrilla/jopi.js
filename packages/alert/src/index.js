import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
