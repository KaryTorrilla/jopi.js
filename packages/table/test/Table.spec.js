import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Table } from '../src'

describe('Table', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Table>
          <Table.Header>
            <Table.HeaderItem>Agency</Table.HeaderItem>
            <Table.HeaderItem>Country</Table.HeaderItem>
            <Table.HeaderItem>Paid</Table.HeaderItem>
          </Table.Header>
          <Table.Row>
            <Table.RowItem>InmoMex</Table.RowItem>
            <Table.RowItem>Mexico</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
          <Table.Row>
            <Table.RowItem>RGM</Table.RowItem>
            <Table.RowItem>Argentina</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
          <Table.Row>
            <Table.RowItem>Urbania</Table.RowItem>
            <Table.RowItem>Peru</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
