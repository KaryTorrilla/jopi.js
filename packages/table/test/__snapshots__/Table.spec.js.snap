// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<table
  className="c0"
>
  <tr
    className="c0"
  >
    <th
      className="c0"
    >
      Agency
    </th>
    <th
      className="c0"
    >
      Country
    </th>
    <th
      className="c0"
    >
      Paid
    </th>
  </tr>
  <tr
    className="c0"
  >
    <td
      className="c0"
    >
      InmoMex
    </td>
    <td
      className="c0"
    >
      Mexico
    </td>
    <td
      className="c0"
    >
      Yes
    </td>
  </tr>
  <tr
    className="c0"
  >
    <td
      className="c0"
    >
      RGM
    </td>
    <td
      className="c0"
    >
      Argentina
    </td>
    <td
      className="c0"
    >
      Yes
    </td>
  </tr>
  <tr
    className="c0"
  >
    <td
      className="c0"
    >
      Urbania
    </td>
    <td
      className="c0"
    >
      Peru
    </td>
    <td
      className="c0"
    >
      Yes
    </td>
  </tr>
</table>
`;
